RELEASING.md

```diff
@@ -1,120 +1,186 @@
+most cases both dependencies will need to be pinned.
+
+```diff
+diff --git a/hack/update-deps.sh b/hack/update-deps.sh
+index b277dd3ff..1989885ce 100755
+--- a/hack/update-deps.sh
++++ b/hack/update-deps.sh
+@@ -32,8 +32,8 @@ VERSION="master"
+ # The list of dependencies that we track at HEAD and periodically
+ # float forward in this repository.
+ FLOATING_DEPS=(
+-  "knative.dev/test-infra@${VERSION}"
+-  "knative.dev/pkg@${VERSION}"
++  "knative.dev/test-infra@release-x.y"
++  "knative.dev/pkg@release-x.y"
+   "knative.dev/caching@${VERSION}"
+ )
+```
 
-> If by mid-week, we do not have a clean PR in each repository, the person
-> driving the update should escalate to avoid delaying the release.
+The downstream repositories this needs to happen on are:
 
-### Step #2: Friday the week prior to each release.
+- [knative/caching](https://github.com/knative/caching)
+- [knative/eventing-contrib](https://github.com/knative/eventing-contrib)
+- [knative/eventing](https://github.com/knative/eventing)
+- [knative/net-certmanager](https://github.com/knative/net-certmanager)
+- [knative/net-contour](https://github.com/knative/net-contour)
+- [knative/net-http01](https://github.com/knative/net-http01)
+- [knative/net-istio](https://github.com/knative/net-istio)
+- [knative/net-kourier](https://github.com/knative/net-kourier)
+- [knative/operator](https://github.com/knative/operator)
+- [knative/serving](https://github.com/knative/serving)
 
-A release branch is snapped on knative/pkg with the form `release-0.X` where `X`
-reflects the forthcoming Knative release.
+Apply the changes the the **master branches**, run
+`hack/update-deps.sh --upgrade` (and potentially `hack/update-codegen.sh` if
+necessary) and PR the changes to the **master branch**. Don't cut the release
+branch yet.
 
-Any releaseable components inside of `knative/pkg` also need to be set to their
-release branch to avoid trivial point releases during the next release. At the
-moment we have the following Knative dependecies inside of `pkg`:
+### Cut and pin `caching`
 
-- knative.dev/test-infra
+`caching` sees very little traffic, so we cut it 1 week prior to the actual
+release to unblock the `serving` release (it depends on caching). Cut a
+`release-x.y` branch and pin it in `serving` as shown above. You can potentially
+collapse this pin with the `pkg`/`test-infra` pin to `serving` per step 2.
 
-Update these to point to a release branch for the current release and update the
-toml entry on the `knative.dev/pkg:release-0.X` branch, like:
+### Verify nightly release automation is intact
 
-```toml
-[[constraint]]
-  name = "knative.dev/test-infra"
-  branch = "release-0.14"
-```
+The automation used to cut the actual releases is the very same as the
+automation used to cut nightly releases. Verify via testgrid that all relevant
+nightly releases are passing. If they are not coordinate with the relevant WG
+leads to fix them.
```

Comment:
@chaodaiG @chizhg it'd be useful if we could simply inline a "badge" for each of the checks called out here.